lib/Config.js

```diff
@@ -751,20 +799,34 @@ class Config extends EventEmitter {
         this.emit('rest-endpoints-update');
     }
 
-    getAzureEndpoint(locationConstraint) {
-        return process.env[`${locationConstraint}_AZURE_BLOB_ENDPOINT`] ||
-        this.locationConstraints[locationConstraint].details.azureBlobEndpoint;
+    getAzureStorageAccountName(locationConstraint) {
+        const { azureStorageAccountName } =
+            this.locationConstraints[locationConstraint].details;
+        const storageAccountNameFromEnv =
+            process.env[`${locationConstraint}_AZURE_STORAGE_ACCOUNT_NAME`];
+        return storageAccountNameFromEnv || azureStorageAccountName;
     }
 
-    getAzureBlobSas(locationConstraint) {
-        return process.env[`${locationConstraint}_AZURE_BLOB_SAS`] ||
-        this.locationConstraints[locationConstraint].details.azureBlobSAS;
+    getAzureStorageCredentials(locationConstraint) {
+        const { azureStorageAccessKey } =
+            this.locationConstraints[locationConstraint].details;
+        const storageAccessKeyFromEnv =
+            process.env[`${locationConstraint}_AZURE_STORAGE_ACCESS_KEY`];
+        return {
+            storageAccountName:
+                this.getAzureStorageAccountName(locationConstraint),
+            storageAccessKey: storageAccessKeyFromEnv || azureStorageAccessKey,
+        };
     }
 
     isSameAzureAccount(locationConstraintSrc, locationConstraintDest) {
-        return locationConstraintDest ?
-        this.getAzureEndpoint(locationConstraintSrc) ===
-        this.getAzureEndpoint(locationConstraintDest) : true;
+        if (!locationConstraintDest) {
+            return true;
+        }
+        const accountNames = [locationConstraintSrc, locationConstraintDest]
```

Comment:
style: for a simple comparison, I think calling twice `getAzureStorageAccountName` and comparing the results would be more straigthtforward than relying on a map.

Reply:
That's true, I updated it